protoc-gen-kotlin/lib/src/commonMain/kotlin/pbandk/gen/CodeGenerator.kt

```diff
@@ -1,10 +1,12 @@
 package pbandk.gen
 
 open class CodeGenerator(val file: File, val kotlinTypeMappings: Map<String, String>, val params: Map<String, String>) {
+    private val jsonSerializerNotGeneratedException: String = "throw UnsupportedOperationException(\"Json serializer generation is disabled. [json_generate_serializer=false]\")"
     protected val bld = StringBuilder()
     protected var indent = ""
 
     protected val jsonUseProtoNames = params["json_use_proto_names"]?.toBoolean() ?: false
+    protected val jsonGenerateSerializer = params["json_generate_serializer"]?.toBoolean() ?: true
```

Comment:
This option controls both JSON serialization _and_ deserialization. So I'd name it more generally, maybe `json_support=true/false`?